astroid/brain/brain_namedtuple_enum.py

```diff
@@ -182,23 +195,34 @@ def _looks_like(node, name):
 _looks_like_typing_namedtuple = functools.partial(_looks_like, name="NamedTuple")
 
 
-def infer_named_tuple(node, context=None):
+def infer_named_tuple(
+    node: nodes.Call, context: Optional[InferenceContext] = None
+) -> Iterator[nodes.ClassDef]:
     """Specific inference function for namedtuple Call node"""
-    tuple_base_name = nodes.Name(name="tuple", parent=node.root())
-    class_node, name, attributes = infer_func_form(
-        node, tuple_base_name, context=context
-    )
+    # Infer which type of NamedTuple we're dealing with (typing or collections)
+    inferred_namedtuple_call = next(node.func.infer())
+    base_names = [
+        nodes.Name(name="tuple", parent=node.root()),
+        nodes.Name(
+            name=inferred_namedtuple_call.name,
+            parent=inferred_namedtuple_call.root(),
+        ),
+    ]
+
+    class_node, name, attributes = infer_func_form(node, base_names, context=context)
     call_site = arguments.CallSite.from_call(node, context=context)
-    node = extract_node("import collections; collections.namedtuple")
-    try:
 
-        func = next(node.infer())
-    except StopIteration as e:
-        raise InferenceError(node=node) from e
     try:
-        rename = next(call_site.infer_argument(func, "rename", context)).bool_value()
+        rename = next(
+            call_site.infer_argument(inferred_namedtuple_call, "rename", context)
+        ).bool_value()
     except (InferenceError, StopIteration):
         rename = False
+    # If inferred_namedtuple_call is the ClassDef of typing.NamedTuple
+    # infer_argument will raise AttributeError
+    # TODO: See if this exception can be prevented
+    except AttributeError:
+        rename = False
 
```

Comment:
Should we also support the parameters `defaults` and `module`? I can give this a shot. Do we need version guards?

https://docs.python.org/3/library/collections.html#collections.namedtuple

Reply:
I think those are both out of scope for this PR, but I think at some point we will want to. Basically this PR is meant as the base for work on stuff like #242 and #182.

I think the biggest open question for this PR is whether creating a new `Instance` sub class is the right way to go here.  Or if there should be another mechanism for this.